### DISCUSSION TITLE
Contribution submission for Exit this page component

### DISCUSSION THREAD

**Discussion by rwk-moj** [alphagov/govuk-design-system#2923] [Exit this page]
Formerly known as 'Hide this page component'

# What

### What are we proposing 
We are looking to implement a component, using the existing button component, that allows vulnerable users of a Government service to quickly and safely hide the page and reduce the risk of Domestic Abuse incidents.

The functionality behind the component would allow a user to quickly hide the current page and open a new, ambiguous webpage, such as BBC Weather or another such website. This would lower the risk of drawing attention to browsing habits from a perpetrator. 

# Why

### Why are we proposing this design pattern?

> Domestic abuse is physical, sexual, psychological or financial abuse that takes place within an intimate or family-type relationship and that forms a pattern of forceful and controlling behaviour. This can include forced marriage and so-called ‘honour-based crimes.” Domestic abuse may include a range of abusive behaviours, not all of which are in themselves inherently ‘violent’

The biggest risk to a victim of Domestic Abuse is homicide and serious physical assaults - this is at its greatest risk when a victim is planning to flee or escape from the perpetrator. Often, one of the first things a victim will do is to speak to someone to make a disclosure about abuse. The second thing they will do is research and find help online, often using Government services.

We need to ensure that the risk is minimised for users when using online services. Although victims and survivors can be aware of the risks, it is the Government's responsibility to protect individuals using a service. Therefore, it is crucial that there is a consistency of approach in design and functionality across Government with this component.

### What evidence do you have that it's needed by multiple services across government?

From conversations between the Department for Work and Pensions, Ministry of Justice, the Cabinet Office and Scottish Government, it is already clear that other services need to ensure that this design is available. 

Services are already responding to the rise in Domestic Abuse during COVID-19 by building their own minimum viable solutions to allow users to quickly hide their webpages.

There are areas of the Government site, such as “Domestic Abuse: How to get help”, that a high risk user would access for support that does not include a way to hide their browsing. This is putting those users of this service at risk.

### What evidence do you have that it meets the needs of the users of those services?

Speaking with subject matter experts (SME), it is critical that we offer this feature on the GOV.UK website. It is a feature that is provided by other organisations, such as:

- [Scottish Government - Domestic Abuse page](https://www.mygov.scot/domestic-abuse/)
- [Refuge](https://www.refuge.org.uk/)
- [Women’s Aid](https://www.womensaid.org.uk/)
- [Victim Support](https://www.victimsupport.org.uk/)

Each website has varied design implementations, but ultimately addresses the same user need of being able to hide browsing habits quickly.

The Ministry of Justice have carried out some initial research with an SME to establish whether the proposed solution meets the user needs. 

### Have you checked that it doesn't already exist in the GOV.UK Design System?

There are no outstanding issues or tickets in the Design System backlog that refer to this type of component. 

# Anything else

### What else has been done 

The MoJ have iterated twice on the design component on our live service - looking at language, design and functionality.

There are several designs that are considered minimum viable products which have been released by the DWP (Child Maintenance Service) - Cabinet Office (Coronavirus Support) and the MoJ (Legal Aid Agency). The expediency of these releases has been due to the fact that by not having something, we are putting our users at risk.

The MoJ have carried out some early rounds of testing with an SME that have looked at design, functionality and various use cases from a range of user scenarios. This has all been documented and shared with DWP and CO to share research and findings. 

### Designs

We looked to extend the GOV.UK Button component to act as the interactive element that a user would press to exit the website.

The Design System suggests using red for any destructive actions undertaken on a GOV.UK website - as the intention for the 'Hide this page' button was essentially closing the webpage down, it would be deemed destructive. This has been adopted by DWP and the Scottish Government.

We decided to utilise the additional space available in the 1/3 column, which is usually reserved for additional information as it allowed us to present the button within immediate view for the user. 

The button would then be "sticky" in position at the top of the page on smaller devices. This prevents any conflict with the GOV.UK 'Contents' link that is positioned at the bottom of the screen. It also removes potential accidental presses by a user who doesn't need to use it.

**Feedback from SME on the design**
>Victims of Domestic Abuse are used to using their reptilian brain and are in fight or flight mode so the red would be a trigger for them to draw their to attention to it - Comment from SME

### Content

The language used in the button has recently been changed in order to create a more unified style as we continue to work with DWP, Cabinet Office and the Scottish Government. The use of 'Hide this page' presented more of a sense of immediacy, as well as being more descriptive of what the button actually does when the user interacts with it, so we have moved away from the original 'Quick Exit'.

We also use a keyboard function for the user to be able to interact with the button which is controlled by pressing the 'Esc' key.

### Behaviour and Functionality

We have explored several types of behaviour for the button, however in order to provide the greatest amount of safety for users we decided to implement functionality similar to other charities and organisations.

When a user interacts with the button, they are sent to a new browser tab that opens BBC Weather. The original tab that they were on doesn't close, but redirects to Google.

This component does not prescribe which websites are to be linked to. That is best done by the individual services and their user research combined with the knowledge of their user base. Google and BBC Weather are good neutral examples pending further research.

### Accessibility

The Legal Aid Agency have had an Accessibility Audit on their service and this highlighted some improvements to the component. These were:

- Ensuring the button text accurately describes it's purpose
- Ensure the instruction "(Or press 'Esc' key)" is presented to all users when presented as a sticky element
- Include visually hidden text at the top of the page to help users of assistive technology be aware of the leave page functionality
- Differentiate betwixt mobile displays and PC displays zoomed in, so the information about using the escape key is not hidden for non-mobile narrow displays

These changes have been implemented in the latest version of the component, however we will be looking to further test the accessibility of the component as we continue to iterate on it.

### MoJ Data Analytics

The 'Hide this page' button has been running as an MVP on the Check Legal Aid service since 14 April 2020. It has been used with a fair amount of frequency, however at present we are are yet to analyse the data in more detail.

The gaps in data recording were due to changes on the back-end of our service that caused GA to drop out. 

**Graph data since 14 April 2020** 

<img width="997" alt="Screenshot 2020-10-29 at 14 17 21" src="https://user-images.githubusercontent.com/57358216/97587209-0cf40c80-19f3-11eb-9ae9-b6020e2f73a3.png">

**The below graph data also shows variations in spikes and lower use through the pandemic**

<img width="1057" alt="Screenshot 2020-10-14 at 14 43 58" src="https://user-images.githubusercontent.com/57358216/97587349-32811600-19f3-11eb-92a5-026e28432d2a.png">

### Examples

MoJ | DWP
------------ | -------------
<img src="https://user-images.githubusercontent.com/57358216/97587450-4e84b780-19f3-11eb-8f7d-952723dba208.png" alt="Legal Aid component design" width="320"/> | <img src="https://user-images.githubusercontent.com/57358216/97587510-60665a80-19f3-11eb-98ba-84f15c666e5c.png" alt="Child Maintenance component design" width="320"/> 

SCOT.GOV | Cabinet Office
------------ | -------------
Image to follow | <img src="https://user-images.githubusercontent.com/57358216/80375782-bf0a1a80-8890-11ea-8dc7-ffb097e70210.png" alt="Scot Gov component design" width="320"/>

**Comment by JennyG303** [comment]
**How should the escape button work?** 
@htmlandbacon documented how various examples are working online.

![QE-myScot](https://user-images.githubusercontent.com/63775571/80493860-d95dfa00-895d-11ea-98e2-e64f2fe9faf8.png)
![QE-Govuk](https://user-images.githubusercontent.com/63775571/80493865-dbc05400-895d-11ea-94e5-c6fce80892d3.png)
![QE-MOJ](https://user-images.githubusercontent.com/63775571/80493872-dd8a1780-895d-11ea-9f95-c51ac4dee722.png)
![QE-MOJ2](https://user-images.githubusercontent.com/63775571/80493876-dfec7180-895d-11ea-9d49-22d6b7350576.png)

**Women’s Aid “Exit site”**
https://www.womensaid.org.uk/

Position: 
desktop and mobile - upper right fixed position during scroll

Action: 
Goes to Google search in same window 
Back in browser goes back to previously viewed Women’s Aid page

<img width="493" alt="Screenshot 2020-05-05 at 13 36 13" src="https://user-images.githubusercontent.com/63775571/81066726-76c4ab00-8ed5-11ea-9719-ee62ff1f9882.png">

**Comment by htmlandbacon** [comment]
Just a couple of things:

- I've not reviewed these so may be slightly wrong
- I've not done any sort of a11y review ( *ahem* @abbott567)
- I've not checked session behaviour

Assumption is that if a user clicked this while in our journey we'd probably scrub the whole session before we send them off @JennyG303

**Comment by abbott567** [comment]
Accessibility would be an interesting one here, particularly if using a screen reader or voice command given that 'hide this page now' would be audible.

**Comment by htmlandbacon** [comment]
Below is an output of the end of a couple of conversations:

- Colour choice of button
- Placement on desktop (i.e below/above right hand content)
- Sticky behaviour of button on desktop and mobile and location
- Link to the BBC weather page vs some sort of session clearance 

### Desktop

<img width="1001" alt="Example: hide this page button on desktop" src="https://user-images.githubusercontent.com/4334015/83248549-23284300-a19d-11ea-839d-bf2c5e00b598.png">

<img width="984" alt="Example: hide this page button on desktop after scrolling" src="https://user-images.githubusercontent.com/4334015/83248530-1a377180-a19d-11ea-8ea5-6230f8f4908b.png">

### Mobile

<img width="494" alt="Example: hide this page button on mobile" src="https://user-images.githubusercontent.com/4334015/83248749-74d0cd80-a19d-11ea-830c-362ceea9e8d3.png">

<img width="499" alt="Example: hide this page button on mobile after scrolling" src="https://user-images.githubusercontent.com/4334015/83248827-903bd880-a19d-11ea-8bde-87d9e99f794e.png">

**Comment by laniknight** [comment]
In relation to the new WCAG 2.4.11 focus not obscured - is there a concern that an always visible exit button would cause a failure on this new criteria?

**Reply by laniknight** [reply]
For reference this is our current implementation https://somersetdomesticabuse.org.uk/ - am keen to keep us compliant when we start testing for 2.2

**Reply by calvin-lau-sig7** [reply]
HI @laniknight, thanks for sharing your implementation.

Yes, we had similar thoughts which we gave brief mention in our rationale on the [button placement](https://github.com/alphagov/govuk-design-system/discussions/2909), and moving it to the likely sparser right side of the page. 

I'm not an accessibility expert, but I reckon it'll depend on how likely content is to fall under the button. Be interested to hear any results from your testing!

**Comment by amyhupe** [comment]
Hello 👋  I'm supporting a client team at a local authority who's looking at implementing this component as part of its service. We have a question and a piece of feedback to share.

1. **Question:** Discussing the component within our team, we though that "exit this site" would be a more accurate button label than "exit this page" - since the button links users to a different website. Is there any reason this label wasn't used in the first place that we should be aware of?

2. **Feedback:** We recently conducted some user research on the implementation and wanted to share that **all** participants we tested with said they believed that the link going to BBC news / weather would raise suspicion with their perpetrator. For this reason, we are going to implement the component with a link to Google.

**Reply by Nooshu** [reply]
> 2\. **Feedback:** We recently conducted some user research on the implementation and wanted to share that **all** participants we tested with said they believed that the link going to BBC news / weather would raise suspicion with their perpetrator. For this reason, we are going to implement the component with a link to Google.

This looks to be the correct decision and is supported by data from [June 2024](https://www.similarweb.com/top-websites/united-kingdom/). Google.com is the most visited website in the United Kingdom in June.

**Reply by querkmachine** [reply]
I have a very particular objection to the EtP button linking to the Google homepage: It isn't a contentful page, it's a landing page specifically designed to be a page that users leave quickly. 

No one using a computer in a natural capacity is going to spend more than a couple of seconds looking at Google's homepage. Even mildly tech savvy people may know how to search via their browser and never visit Google's homepage.

In my opinion, directing an at-risk user to the homepage of a search engine is more likely to raise suspicion in an observer than a page a user would normally spend time browsing, such as that of a news website (or, given we risk accusations of bias doing that) the BBC Weather page that we settled upon.

**Reply by Nooshu** [reply]
What about a combination of both? Link to the Google search page that displays BBC News or BBC weather results?

https://www.google.co.uk/search?q=bbc+news
https://www.google.co.uk/search?q=bbc+sport
https://www.google.co.uk/search?q=bbc+weather (I have a concern with the weather page as it shows an additional popup asking the user for their location, this could be confusing to the user).

Or another alternative is linking to YouTube, since it is both contentful and the 2nd most popular site in the UK for June

<img width="777" alt="Screenshot 2024-07-31 at 16 07 42" src="https://github.com/user-attachments/assets/0ad586f0-92b7-4ccc-b1c4-9539bf6abe28">
